fscryptctl.c

```diff
@@ -440,7 +440,7 @@ static int cmd_remove_key(int argc, char *const argv[]) {
   const char *key_identifier = argv[0];
   const char *mountpoint = argv[1];
 
-  struct fscrypt_remove_key_arg arg = {};
+  struct fscrypt_remove_key_arg arg = {0};
```

Comment:
I'm not sure this way is better, since the new way is arguably missing a set of braces due to the first member of the struct itself being a struct.  We've had to fix up this some thing in kernel code previously, except it was the other way around -- old compilers were warning about `{0}`, so it had to be changed to `{}` (e.g. https://lkml.kernel.org/linux-scsi/20201002063538.1250-1-shipujin.t@gmail.com/).

Is `{0}` fully C99-compliant, in the sense that if it generates a compiler warning, we can just say that the compiler needs to be upgraded?  `fscryptctl` doesn't necessarily have to support compilers as old as the Linux kernel, I suppose...

Reply:
Looking at: https://en.cppreference.com/w/c/language/struct_initialization#Nested_initialization

> If the nested initializer does not begin with an opening brace, only enough initializers from the list are taken to account for the elements or members of the member array, struct or union; any remaining initializers are left to initialize the next struct member:
```c
struct example {
    struct addr_t {
       uint32_t port;
    } addr;
    union {
       uint8_t a8[4];
       uint16_t a16[2];
    } in_u;
};

struct example ex = {80, 127, 0, 0, 1}; // 80 initializes ex.addr.port
                                        // 127 initializes ex.in_u.a8[0]
                                        // 0 initializes ex.in_u.a8[1]
                                        // 0 initializes ex.in_u.a8[2]
                                        // 1 initializes ex.in_u.a8[3]
```

So `{0}` is always good C99.

Reply:
Both GCC (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53119) and Clang (https://bugs.llvm.org/show_bug.cgi?id=21689) said this was a bug, so I think we're OK.

Reply:
Okay, the clang fix is more recent than I expected, but I suppose we should just use `{0}`.